docs/birthday-kata.md

```diff
@@ -0,0 +1,890 @@
+```
+
+This is another mock to make it easy to test.
+1) This handler takes a list of employees, which it uses to generate the return value for `fetchAllByBirthday`.
+2) Our `h` take a request with the `EmployeeRepository` ability.
+3) We match on the `fetchAllByBirthday` method and it's argument `birthday`.
+4) When we match on that method, we filter the provided list of employees based on the `birthday` arg, and resume the computation with those employees as the return value.
+5) In real life, we might read from a DB here.
+6) Otherwise we bubble up the function in case other handlers know how to handle it.
+
+
+And finally
+```unison
+MessageService.handler.mock : k -> k
+MessageService.handler.mock k =
+  h : Request {MessageService} k -> k
+  h = cases
+    { send message -> resume } -> handle resume message with h
+    { resume } -> resume
+  handle k with h
+```
+
+1) This handler does not take any args because it doesn't need any outside info to handle the `send` function.
+2) We match on the `send` function and its message argument and resume the computation with that same message as a return value
+3) A real version of this might instead reach out to Twilio or some external service to perform an effect before continuing.
+4) And we bubble up any functions we don't care about.
+
+Great! And then we just wrap all these handlers around our birthday function.
+```unison
+test> sendBirthdayEmails.tests.noBirthdaysToday = 
+  check let
+    actual = 
+      handle 
+        handle 
+          handle 
+            !sendBirthdayEmails 
+          with (Calendar.handler.mock (Date (Day 1) January (Year 1991)))
+        with (EmployeeRepository.handler.mock [])
+      with MessageService.handler.mock
+    expected = []
+    if expected === actual
+    then true
+    else bug (expected, actual)
+```
+
+And thats how we do "dependency injection" in unison! We can easily swap out these handlers with other behaviors at the boundary of our system!
+
+Now that we have a passing test, let's take a minute to refactor a bit before we add some more interesting test cases.
+
+All these nested handlers are kind of gross. Let's combine them into a single "multihandler".
+```unison
+sendBirthdayEmails.handlers.mock : Date -> [Employee] -> k -> k
+sendBirthdayEmails.handlers.mock date employees k =
+  h : Request {Calendar, EmployeeRepository, MessageService} k -> k
+  h = cases
+    {today -> resume} -> handle resume date with h
+    { send message -> resume } -> handle resume message with h
+    { fetchAllByBirthday birthday -> resume } ->
+      birthdays = filter (employee -> (Date.day (Employee.birthday employee), Date.month (Employee.birthday employee)) === birthday) employees
+      handle resume birthdays with h
+    { resume } -> resume
+  handle k with h
+
+test> sendBirthdayEmails.tests.noBirthdaysToday = 
+check let
+  actual = 
+    handle 
+      !sendBirthdayEmails 
+    with sendBirthdayEmails.handlers.mock (Date (Day 1) January (Year 1991)) []
```

Comment:
Love this simplified handler! If you delay the computation `sendBirthdayEmails.handlers.mock : Date -> [Employee] -> '{Calendar, EmployeeRepository, MessageService} k -> k` you don't need to use the `handle` syntax in the actual test.  `actual = sendBirthdayEmails.handlers.mock (Date (Day 1) January (Year 1991)) [] sendBirthdayEmails` <-- this is more idiomatic for calling a handler.